flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1356,6 +1361,61 @@ IntrinsicLibrary::genPresent(mlir::Type,
                                           fir::getBase(args[0]));
 }
 
+// SCAN 
+fir::ExtendedValue
+IntrinsicLibrary::genScan(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() >= 2);
+
+  // required string argument
+  auto string = builder.createBox(loc, args[0]);
+  // required set argument
+  auto set = builder.createBox(loc, args[1]);
+
+  // Handle optional argument, kind
+  // TBD: Where to get default integer kind? For now, use 4.
+  auto kind = fir::getBase(args[3]) ? fir::getBase(args[3]) :
+              builder.createIntegerConstant(loc, resultType, 4);
+  
+  // Create result descriptor    
+  auto resultMutableBox =
+       Fortran::lower::createTempMutableBox(builder, loc, resultType);
+  auto resultIrBox =
+      Fortran::lower::getMutableIRBox(builder, loc, resultMutableBox);
+
+
+  // Handle optional back argument then call Fortran::lower::genScan().
+  if (!fir::getBase(args[2])) {
+    // back argument not specified, so pass in default logical false by
+    // passing in default integer 0.
+    // TBD: Is there a better way to pass in a constant logical false?
```

Comment:
For what you want to do, not. There is `builder.createBool()` but it creates `i1`, which is what is needed in fir.if/fir.select and all the branching operations. But to create a Fortran logical we do not have a helper, so what you have here is pretty much OK (a nit would be to use the fir::LogicalType).

However, I think you do not need to do all that given the runtime interface takes `back` as an optional pointer argument and will deduce by itself that it is false if not present.

Which will lead you to create something I did not nicely document, which is how to create an "absent" decsriptor to pass to the runtime.

You need to use the fir.absent op with the a fir.box type, fir.box<None> is just fine. You can do something like:
```
  auto boxType = fir::BoxType::get(builder.getNoneType()) ;
  auto backBox= builder.create<fir::Absent>(loc, builder.getNoneType());
```